### PR TITLE
Add an API to get the canvas / map coordinate in the context menu callback.

### PR DESCRIPTION
**Pull request by ThatRedox**
This sets the user data of the context menu to the `Vector2` representing the coordinate.

### CODE REVIEW

File: chunky/src/java/se/llbit/chunky/ui/RenderCanvasFx.java [hunk 1]
```diff
@@ -247,6 +247,7 @@ public RenderCanvasFx(ChunkyFxController chunkyFxController,
         double halfWidth = canvas.getWidth() / (2.0 * canvas.getHeight());
         target.set(-halfWidth + event.getX() * invHeight,
             -0.5 + event.getY() * invHeight);
+        contextMenu.setUserData(new Vector2(event.getX(), event.getY()));
```
Comment (NotStirred) [inline chunky/src/java/se/llbit/chunky/ui/RenderCanvasFx.java:250]: Any reason not to use properties? imo it's more readable
```suggestion
        contextMenu.getProperties().put("canvasPos", new Vector2(event.getX(), event.getY()));
```

Comment (ThatRedox) [reply chunky/src/java/se/llbit/chunky/ui/RenderCanvasFx.java]: Didn't know you could do that 👍

File: chunky/src/java/se/llbit/chunky/main/Chunky.java [hunk 2]
```diff
@@ -479,6 +479,7 @@ public TabTransformer getMainTabTransformer() {
 
   /**
    * Get the mutable list of context menu transformers in the main map view.
+   * The map view X and Y coordinates are supplied as a `Vector2` in the context menu's `mapviewPosition` property.
```
Comment (ShirleyNekoDev) [inline chunky/src/java/se/llbit/chunky/main/Chunky.java:482]: Could you please clarify the unit for the coordinate system. Is it in pixels of the FX pane or in Minecraft blocks? Ideally I'd like to have access to both in different properties :)

File: chunky/src/java/se/llbit/chunky/main/Chunky.java [hunk 3]
```diff
@@ -488,6 +489,7 @@ public List<ContextMenuTransformer> getMapContextMenuTransformers() {
 
   /**
    * Get the mutable list of context menu transformers in the main render view.
+   * The canvas X and Y coordinates are supplied as a `Vector2` in the context menu's `canvasPosition` property.
```
Comment (ShirleyNekoDev) [inline chunky/src/java/se/llbit/chunky/main/Chunky.java:492]: Also, could you please clarify the origin for the coordinate system (which corner) and if it is for the surrounding FX element or the canvas itself (and how does scaling affect these? Do you get the pixel location for lookup in the render buffer or something on which you have conversion to do?).

**Review by ShirleyNekoDev — CHANGES_REQUESTED** [review]

File: chunky/src/java/se/llbit/chunky/main/Chunky.java (hunk 2 above)
Comment (ThatRedox) [reply chunky/src/java/se/llbit/chunky/main/Chunky.java]: Looks like its in pixels of the FX pane. I updated the Javadocs and also added a property with the chunk coordinates (multiply by 16 to get blocks).

File: chunky/src/java/se/llbit/chunky/main/Chunky.java (hunk 3 above)
Comment (ThatRedox) [reply chunky/src/java/se/llbit/chunky/main/Chunky.java]: It's the coordinate of the canvas which is the same as the pixel location on the render buffer. I updated the javadocs.

**Review by ShirleyNekoDev — APPROVED** [review]

**Review by NotStirred — APPROVED** [review]
New additions look great :+1:

**Comment by NotStirred** [comment]
Thoughts on adding a comment next to any added properties that they're PluginAPI?

**Review by leMaik — APPROVED** [review]

**Pull request merged**